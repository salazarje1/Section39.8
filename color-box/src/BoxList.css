.BoxList-boxes {
    display: flex; 
    flex-direction: column;
}