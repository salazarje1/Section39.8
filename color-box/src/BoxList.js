import React, { useState } from 'react'; 
import { v4 as uuid } from 'uuid'
import Box from './Box';
import NewBoxForm from './NewBoxForm';

const BoxList = () => {
    const INITIALSTATE = [
        {id: uuid(), color: "red", width: 10, height: 10},
        {id: uuid(), color: "blue", width: 3, height: 3}
    ]; 

    const [boxes, setBoxes] = useState(INITIALSTATE); 

    const addBox = (newBox) => {
        setBoxes(boxes => [...boxes, { ...newBox, id: uuid() }])
    }

    const removeBox = (boxId) => {
        setBoxes(boxes => boxes.filter(box => box.id !== boxId)); 
    }

    return (
        <div className='BoxList'>
            <h2>Boxes</h2>
            <NewBoxForm addBox={addBox} />
            <div className='BoxList-boxes'>
                {boxes.map(({id, color, width, height}) => <Box removeBox={removeBox} id={id} key={id} color={color} width={width} height={height} />)}
            </div>
        </div>
    )
}

export default BoxList; 