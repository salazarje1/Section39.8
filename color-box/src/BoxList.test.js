import { render, fireEvent } from '@testing-library/react'; 
import BoxList from './BoxList';

test('box list renders without crashing', () => {
    render(<BoxList />);
})

test('it matches snapshop', () => {
    const { asFragment } = render(<BoxList />)

    expect(asFragment()).toMatchSnapshot(); 
})

test('form should add new box to list', () => {
    const { container, getByLabelText, queryByText } = render(<BoxList />);

    const boxes = container.getElementsByClassName('Parent');
    const input = getByLabelText("Color");
    const btn = queryByText('Add Box'); 

    expect(boxes.length).toBe(2); 

    fireEvent.change(input, { target: { value: 'Green' } }); 
    fireEvent.click(btn); 

    expect(boxes.length).toBe(3); 

})