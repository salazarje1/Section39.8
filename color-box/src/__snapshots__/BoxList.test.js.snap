// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches snapshop 1`] = `
<DocumentFragment>
  <div
    class="BoxList"
  >
    <h2>
      Boxes
    </h2>
    <form>
      <div>
        <label
          for="color"
        >
          Color
        </label>
        <input
          id="color"
          name="color"
          placeholder="color"
          type="text"
          value=""
        />
      </div>
      <div>
        <label
          for="width"
        >
          Width
        </label>
        <input
          id="width"
          max="10"
          min="1"
          name="width"
          placeholder="width"
          type="number"
          value=""
        />
      </div>
      <div>
        <label
          for="height"
        >
          height
        </label>
        <input
          id="height"
          max="10"
          min="1"
          name="height"
          placeholder="height"
          type="number"
          value=""
        />
      </div>
      <button>
        Add Box
      </button>
    </form>
    <div
      class="BoxList-boxes"
    >
      <div
        class="Parent"
      >
        <div
          class="Box"
          style="background-color: red; width: 200px; height: 200px;"
        />
        <button
          class="Box-button"
          style="background-color: red;"
        >
          X
        </button>
      </div>
      <div
        class="Parent"
      >
        <div
          class="Box"
          style="background-color: blue; width: 60px; height: 60px;"
        />
        <button
          class="Box-button"
          style="background-color: blue;"
        >
          X
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
